dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/WebContainer.java

```diff
@@ -390,6 +392,14 @@ public void deactivate(ComponentContext componentContext) {
         if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
             Tr.event(tc, "Deactivating the WebContainer bundle");
         }
+       
+        //issue#24461
+        if (FrameworkState.isStopping() && AsyncContextImpl.executorRetrieved.get()) {
```

Comment:
Don't need `!AsyncContextImpl.ExecutorFieldHolder.field.isShutdown()` in the conditional any longer? 

Maybe update the issue number since you are switching to a new issue.

Reply:
I do wonder why we are using a separate executor instead of using the global scheduled executor.

Reply:
> Don't need `!AsyncContextImpl.ExecutorFieldHolder.field.isShutdown()` in the conditional any longer?

Once the executorRetrieved.set() turns true, the `field` can be idle but never shutdown til the server is shutdown.  Thus I don't think conditional check .isShutdown() is needed anymore.
> 
> Maybe update the issue number since you are switching to a new issue.
Updated